compiler/resolution/wrappers.cpp

```diff
@@ -1627,6 +1628,24 @@ static void handleInIntent(FnSymbol* fn, CallExpr* call,
       // (don't destroy it here, it will be destroyed there).
       actualSym->addFlag(FLAG_NO_AUTO_DESTROY);
 
+      // domain literals are constant by definition. If we are moving a domain
+      // literal into an `in` formal, we don't have any initCopy etc to fix
+      // constness. So, adjust the call that set that domain to be constant
```

Comment:
I would think that this code should apply to any function returning a domain rather than just `chpl__buildDomainExpr`. Why are domain literals special in this regard? Could we create another example with a function returning a domain that it declared as `const` that is then stored in a `var`? Does that case work correctly before this PR? (If so, maybe we should be doing something to adjust `chpl__buildDomainExpr` to work more similarly to these other calls).

You could extend your new test to include checking a function that returns a domain (like `proc makeConstDomain() { return {1..3}; }` ) to investigate this.